### ISSUE TITLE
Azureml-mflow authentication fails on compute cluster when using nested parallel runs with joblib

### ISSUE THREAD

**Issue by niberger** [Azure/azure-sdk-for-python#38705]
- **Package Name**: azureml-mlflow
- **Package Version**: 1.58.0.post3
- **Operating System**: ubuntu
- **Python Version**: 3.12

**Describe the bug**
MLFlow support the log of nested parallel runs (e.g. https://github.com/mlflow/mlflow/blob/master/examples/hyperparam/search_random.py).
But on Azure ml compute cluster, this will lead to an authentication error.

**To Reproduce**
1. Submit the following python script on a ml.azure.com compute cluster without any managed identity:
```
import mlflow
from joblib import Parallel, delayed

def child_job(idx):
    with mlflow.start_run(nested=True):
        mlflow.log_metric("val_loss", 0.001)

def main_job(n_jobs = 2):
    with mlflow.start_run():
        Parallel(n_jobs=n_jobs)(delayed(child_job)(idx) for idx in range(n_jobs))

if __name__ == "__main__":
    main_job()
````
2. Wait for the run to fail
3. The std_log.txt will start with: `ChainedTokenCredential failed to retrieve a token from the included credentials.`

Note: if you use a compute cluster with system-assigned managed identity, it will fails instead with the following error:
`mlflow.exceptions.MlflowException: API request to endpoint /api/2.0/mlflow/experiments/get-by-name failed with error code 403 != 200. Response body: ''`

**Expected behavior**
Runs smoothly without any exception, with two `val_loss` metric logged, like it does on a compute node with SSO activated.

**Additional context**
The reason of the exception is a conflict with the behaviour of `mlflow.start_run`.
The main job is able to authenticate with an `AzureMLTokenAuthentication`, because the cluster environment contains a `MLFLOW_TRACKING_TOKEN` as well as a `MLFLOW_RUN_ID` and other env variables.
But `mlflow.start_run()` will then remove `MLFLOW_RUN_ID` from the environment: https://github.com/mlflow/mlflow/blob/402cd0b67d52ae028818125fee8003a3ee0c03c2/mlflow/tracking/fluent.py#L356

When joblib's backend loky is duplicating the env variable for the child process, `MLFLOW_RUN_ID` is missing. The `azureml-mlflow` of the child process won't even try to create a `AzureMLTokenAuthentication`. Instead it will fallback to a `DefaultAzureCredential` that will lead to the exception.

A possible workaround for the user is to add `os.environ["MLFLOW_RUN_ID"] = mlflow.active_run().info.run_id` just after the `mlflow.start_run()`.

**Comment by corticalstack** [comment]
Was having issue executing from a bash step of Azure DevOps pipeline a python script using mlflow with tracking uri to retrieve metrics from an Azure Machine Learning job. Despite setting the documented environment variables for mlflow as pere [here](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-use-mlflow-configure-tracking?view=azureml-api-2&tabs=python%2Cenviron#configure-authentication). azureml-mlflow was always falling into interactive logon authentication in get_service_context_from_tracking_url_default_auth(parsed_url).

Upgraded azureml-mlflow to 1.59.0.post1 fixed issue.
